apps/network-alerts/test/unit/attention/notify_test.js

```diff
@@ -0,0 +1,244 @@
+    navigator.mozSettings = MockNavigatorSettings;
+
+    this.sinon.stub(navigator, 'vibrate');
+    this.sinon.spy(window, 'AudioContext');
+    this.sinon.stub(
+      MockAudioContext.prototype, 'createOscillator',
+      () => new MockOscillatorNode()
+    );
+
+    this.sinon.stub(MockOscillatorNode.prototype, 'connect');
+    this.sinon.stub(MockOscillatorNode.prototype, 'start');
+    this.sinon.stub(MockOscillatorNode.prototype, 'stop');
+
+    this.sinon.stub(
+      MockAudioContext.prototype, 'createGain',
+      () => new MockGainNode()
+    );
+    this.sinon.stub(MockGainNode.prototype, 'connect');
+    this.sinon.stub(MockAudioParam.prototype, 'setValueCurveAtTime');
+
+    getStub = sinon.stub();
+    setStub = sinon.stub();
+
+    this.sinon.stub(MockNavigatorSettings, 'createLock').returns({
+      get: getStub,
+      set: setStub
+    });
+
+    Object.defineProperty(document, 'hidden', {
+      configurable: true,
+      get: () => isDocumentHidden
+    });
+  });
+
+  teardown(function() {
+    navigator.mozSettings = realMozSettings;
+    delete document.hidden;
+  });
+
+  function sendVisibilityEvent() {
+    var visibilityEvent = new CustomEvent(
+      'visibilitychange',
+      { bubbles: true }
+    );
+
+    window.dispatchEvent(visibilityEvent);
+  }
+
+  function assertVibrate() {
+    sinon.assert.calledWith(
+      navigator.vibrate,
+      [2000, 500, 1000, 500, 1000, 500, 2000, 500, 1000, 500, 1000, 500]
+    );
+  }
+
+  function assertAudio() {
+    sinon.assert.calledWith(AudioContext, 'notification');
+    sinon.assert.calledTwice(AudioContext.prototype.createOscillator);
+    sinon.assert.calledOnce(AudioContext.prototype.createGain);
+
+    var audioCtx = AudioContext.firstCall.returnValue;
+    var gainNode = AudioContext.prototype.createGain.firstCall.returnValue;
+    var o1 = AudioContext.prototype.createOscillator.firstCall.returnValue;
+    var o2 = AudioContext.prototype.createOscillator.secondCall.returnValue;
+
+    assert.equal(o1.type, 'sine');
+    assert.equal(o2.type, 'sine');
+
+    // either one case or the other
+    try {
+      assert.equal(o1.frequency.value, 853);
+      assert.equal(o2.frequency.value, 960);
+    } catch(e) {
+      assert.equal(o2.frequency.value, 853);
+      assert.equal(o1.frequency.value, 960);
+    }
+
+    sinon.assert.calledOn(o1.connect, o1);
+    sinon.assert.calledOn(o2.connect, o2);
+    sinon.assert.alwaysCalledWith(o1.connect, gainNode);
+    sinon.assert.calledWith(gainNode.connect, audioCtx.destination);
+
+    sinon.assert.calledOn(o1.start, o1);
+    sinon.assert.calledOn(o2.start, o2);
+    sinon.assert.calledOn(o1.stop, o1);
+    sinon.assert.calledOn(o2.stop, o2);
+    sinon.assert.alwaysCalledWith(o1.stop, 11);
+
+    sinon.assert.calledOn(gainNode.gain.setValueCurveAtTime, gainNode.gain);
+    sinon.assert.calledWith(
+      gainNode.gain.setValueCurveAtTime,
+      sinon.match.instanceOf(Float32Array), 0, 11
+    );
```

Comment:
Honestly if it was SMS code, I'd say yeah it's too verbose and touches internal details too much, but at the same time - maybe it's good for such life-critical (literally) app as CMAS - so that we're sure as much as possible that we play this tone :) So I'm totally fine if you decide to leave it as is.